frontend/packages/integration-tests-cypress/tests/crud/k8-openshift-cruds.spec.ts

```diff
@@ -0,0 +1,171 @@
+    .set('statefulsets', { kind: 'StatefulSet' })
+    .set('resourcequotas', { kind: 'ResourceQuota' })
+    .set('limitranges', { kind: 'LimitRange' })
+    .set('horizontalpodautoscalers', { kind: 'HorizontalPodAutoscaler' })
+    .set('networkpolicies', { kind: 'NetworkPolicy' })
+    .set('roles', { kind: 'Role' });
+  const openshiftObjs = OrderedMap<string, { kind: string; namespaced?: boolean }>()
+    .set('deploymentconfigs', { kind: 'DeploymentConfig' })
+    .set('buildconfigs', { kind: 'BuildConfig' })
+    .set('imagestreams', { kind: 'ImageStream' })
+    .set('routes', { kind: 'Route' })
+    .set('user.openshift.io~v1~Group', { kind: 'user.openshift.io~v1~Group', namespaced: false });
+  const serviceCatalogObjs = OrderedMap<string, { kind: string; namespaced?: boolean }>().set(
+    'clusterservicebrokers',
+    {
+      kind: 'servicecatalog.k8s.io~v1beta1~ClusterServiceBroker',
+      namespaced: false,
+    },
+  );
+
+  let testObjs = Cypress.env('openshift') === true ? k8sObjs.merge(openshiftObjs) : k8sObjs;
+  testObjs = Cypress.env('servicecatalog') === true ? testObjs.merge(serviceCatalogObjs) : testObjs;
+  const testLabel = 'automated-test-name';
+  const resourcesWithCreationForm = new Set(['StorageClass', 'Route', 'PersistentVolumeClaim']);
+
+  testObjs.forEach(({ kind, namespaced = true }, resource) => {
+    describe(kind, () => {
+      const name = `${testName}-${_.kebabCase(kind)}`;
+
+      const createResource = () => {
+        cy.log(`Create ${kind}: ${name}`);
+        cy.visit(
+          `${namespaced ? `/k8s/ns/${testName}` : '/k8s/cluster'}/${resource}?name=${testName}`,
+        );
+        if (kind === 'Secret') {
+          listPage.clickCreateYAMLdropdownButton();
+        } else {
+          listPage.clickCreateYAMLbutton();
+        }
+        if (resourcesWithCreationForm.has(kind)) {
+          cy.byTestID('yaml-link').click();
+        }
+        yamlEditor.isLoaded();
+        let newContent;
+        // get, update, and set yaml editor content.
+        return yamlEditor.getEditorContent().then((content) => {
+          newContent = _.defaultsDeep(
+            {},
+            { metadata: { name, labels: { [testLabel]: testName } } },
+            safeLoad(content),
+          );
+          yamlEditor.setEditorContent(safeDump(newContent, { sortKeys: true }));
+          cy.log('creates a new resource instance');
+          yamlEditor.clickSaveCreateButton();
+          cy.get(errorMessage).should('not.exist');
+        });
+      };
+
+      const deleteResource = () => {
+        cy.log(`Delete ${name}`);
+        cy.visit(`${namespaced ? `/k8s/ns/${testName}` : '/k8s/cluster'}/${resource}`);
+        listPage.filter.byName(name);
+        listPage.rows.countShouldBe(1);
+        listPage.rows.clickKebabAction(name, deleteHumanizedKind(kind));
+        modal.shouldBeOpened();
+        modal.submit();
+        modal.shouldBeClosed();
+        if (kind !== 'user.openshift.io~v1~Group') {
```

Comment:
Don't skip `Group`. Instead check if the string is in `group~version~Kind` format, split on the `~` to get the kind, and lowercase. The command should look like `oc get groups.v1.useropenshift.io`.

I'd create a utility for this and call it from `resourceShouldBeDeleted`. This is untested, but it should be something like

```ts
import { plural } from 'pluralize';

const toCLIType = (type: K8sResourceKindReference | string): string => {
  if (!type?.includes('~')) {
    return type;
  }
  const [group, version, kind] = type.split('~');
  // Resources aren't required to follow this pattern when converting from kind to plural,
  // but this should work for most resources.
  return `${plural(kind.toLowerCase())}.${version}.${group}`;
};
```

Reply:
Hi @spadgett, thanks for the code!  Implemented it in this last [commit](https://github.com/openshift/console/pull/4831/commits/21a0924457e5541bfd2c835cafda4268a1739aae).